yt_dlp/extractor/canvas.py

```diff
@@ -276,35 +276,49 @@ def _login(self):
         if username is None:
             return
 
-        auth_data = {
-            'APIKey': self._APIKEY,
-            'targetEnv': 'jssdk',
-            'loginID': username,
-            'password': password,
-            'authMode': 'cookie',
-        }
-
-        auth_info = self._gigya_login(auth_data)
+        # 1. Get Login Data
+        try:
+            auth_info = self._download_json(
+                'https://accounts.vrt.be/accounts.login', None,
+                note='Login data', errnote='Could not get Login data',
+                headers={}, data=urlencode_postdata({
+                    'loginID': username,
+                    'password': password,
+                    'sessionExpiration': '-2',
+                    'APIKey': self._APIKEY,
+                    'targetEnv': 'jssdk',
+                }))
+        except ExtractorError as e:
+            self.report_warning("Cookie request HTTP code: %s" % e.cause.code)
+            raise e
 
         # Sometimes authentication fails for no good reason, retry
         login_attempt = 1
         while login_attempt <= 3:
             try:
-                # When requesting a token, no actual token is returned, but the
-                # necessary cookies are set.
+                # get XSRF Token
+                self._request_webpage('https://token.vrt.be/vrtnuinitlogin',
+                                      None, note='Requesting XSRF Token', errnote='Could not get XSRF Token',
+                                      query={'provider': 'site', 'destination': 'https://www.vrt.be/vrtnu/'})
+
+                post_data = {
+                    'UID': auth_info['UID'],
+                    'UIDSignature': auth_info['UIDSignature'],
+                    'signatureTimestamp': auth_info['signatureTimestamp'],
+                    'client_id': 'vrtnu-site',
+                }
+                for cookie in self._downloader.cookiejar:
+                    if cookie.name == 'OIDCXSRF':
+                        post_data['_csrf'] = cookie.value
+                        break
```

Comment:
Use `self._get_cookies`

Reply:
I tried to get the `self._get_cookies()` working, but it seems like it's not retrieving the OIDCXSRF cookie. Not sure what's the difference is with the `_request_webpage`.

Reply:
3bc7d5b8d82911fa1cf3a4c42e4c3989b589cf82 should work. Could either of you test?

Reply:
Thanks for your feedback.
I tried a few variants, but all return _401 Authentication failed_ (catched by the `except` clause).
e.g. with the query added: 
`self._get_cookies(update_url_query('https://token.vrt.be/vrtnuinitlogin', 
                                                query={'provider': 'site', 'destination': 'https://www.vrt.be/vrtnu/'}))` 

There seems to be some difference between how `self._request_webpage` (=OK) and `self._get_cookies` (=NOK) trigger the request. Perhaps the `_get_cookies` is not keeping the cookies from the previous `self._download_json('https://accounts.vrt.be/accounts.login', ...` ? Not sure how I can simply log all the request details?

Reply:
`_get_cookies` is not sending any request. It is simply accessing the cookies of the given URL from the cookiejar

Use `--print-traffic` to see the requests. Maybe the cookies are being set for a different domain than the one I set?

Reply:
My apologies for my misunderstanding. I fixed it now to use the _get_cookies() method instead of the internal cookiejar. 

@SamDecrock, can you update your PRQ with the latest from  ytdl-org/youtube-dl#29614?